### PR TITLE
feat: polish starter project code style

### PR DESCRIPTION
**Pull request by ofey404**
I also added more guiding comments in `train.sh`.

image

### CODE REVIEW

File: colossalai_platform/cli/commands/projects/train.py [hunk 1]
```diff
@@ -1,38 +1,74 @@
...
+    )
 
+    # optional arguments, add more if you need
+    parser.add_argument(
+        "--model_dir",
+        type=str,
+        default=None,
+        help="The directory contains the model to finetune.",
+    )
+    return parser.parse_args()
```
Comment (FrankLeeeee) [inline colossalai_platform/cli/commands/projects/train.py:51]: Do not run parse_args for the user, sometimes, they may want to do it on their own. Only adding arguments to the parser will be sufficient.

File: colossalai_platform/cli/commands/projects/train.py [hunk 2]
```diff
@@ -1,38 +1,74 @@
...
-patch_tensorboard()
 
-# Environment variables for your data access
-PROJECT_DIR = os.environ['PROJECT_DIR']
-DATASET_DIR = os.environ['DATASET_DIR']
-OUTPUT_DIR = os.environ['OUTPUT_DIR']
-INPUT_MODEL_DIR = os.environ['MODEL_DIR']
-CHECKPOINT_DIR = os.path.join(OUTPUT_DIR, 'checkpoint')
-TENSORBOARD_DIR = os.path.join(OUTPUT_DIR, 'tensorboard')
+def parse_args(parser: argparse.ArgumentParser):
```
Comment (FrankLeeeee) [inline colossalai_platform/cli/commands/projects/train.py:17]: Name the function name as `add_platform_args`.

File: colossalai_platform/cli/commands/projects/train.py [hunk 3]
```diff
@@ -1,38 +1,74 @@
...
+    # required arguments
+    parser.add_argument(
+        "--epoch",
+        type=int,
+        required=True,
+        help="The training epoch.",
+    )
```
Comment (FrankLeeeee) [inline colossalai_platform/cli/commands/projects/train.py:24]: `epoch` is not a default arg, we should only handle args which are passed as environment variables by the platform. Some programs do not even have epochs, they just set the number of iterations directly.

File: colossalai_platform/cli/commands/projects/train.py (hunk 1 above)
Comment (FrankLeeeee) [inline colossalai_platform/cli/commands/projects/train.py:51]: `tensorboard_dir` is missing

File: colossalai_platform/cli/commands/projects/train.py (hunk 2 above)
Comment (ofey404) [reply colossalai_platform/cli/commands/projects/train.py:17]: Same.

File: colossalai_platform/cli/commands/projects/train.py (hunk 1 above)
Comment (ofey404) [reply colossalai_platform/cli/commands/projects/train.py:51]: Good.

File: colossalai_platform/cli/commands/projects/train.py (hunk 3 above)
Comment (ofey404) [reply colossalai_platform/cli/commands/projects/train.py:24]: But the default `Hyperparameter.json` has a `epoch` field.

This argument is not a default one, but to act as an example:

```json
{
  "HyperParameters": [
    {
      "name": "epoch",
      "types": "string",
      "defaultValue": "10"
    }
  ]
}
```

File: colossalai_platform/cli/commands/projects/train.py (hunk 1 above)
Comment (ofey404) [reply colossalai_platform/cli/commands/projects/train.py]: Updated.

File: colossalai_platform/cli/commands/projects/HyperParameters.json [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
-    {
-      "name": "epoch",
-      "types": "string",
-      "defaultValue": "10"
-    }
+    // TODO: Add your parameters like this:
+    // {
+    //   "name": "epoch",
+    //   "types": "string",
+    //   "defaultValue": "10"
```
Comment (FrankLeeeee) [inline colossalai_platform/cli/commands/projects/HyperParameters.json:7]: Add the "description" field

Comment (ofey404) [reply colossalai_platform/cli/commands/projects/HyperParameters.json]: Done.

**Pull request merged**